snapcraft/commands/central.start

```diff
@@ -1,67 +1,15 @@
 #!/bin/sh
-set -eux
 
-# Load the environment
-. "${SNAP_COMMON}/data/ovn.env"
-
-# Setup directories
-export OVN_DBDIR="${SNAP_COMMON}/data/central/db"
-export OVN_LOGDIR="${SNAP_COMMON}/logs"
-export OVN_RUNDIR="${SNAP_COMMON}/run/ovn"
-export OVN_PKGDATADIR="${SNAP}/share/ovn"
-export OVN_SYSCONFDIR="${SNAP}/etc"
-export OVN_PKIDIR="${SNAP_COMMON}/data/pki"
-
-# Disable some commands
-mkdir -p "${OVN_RUNDIR}/bin/"
-for i in install plymouth sudo systemctl; do
-    [ -e "${OVN_RUNDIR}/bin/${i}" ] && continue
-
-    ln -s "/bin/true" "${OVN_RUNDIR}/bin/${i}"
```

Comment:
Good riddance!

However, I see these are also in `snapcraft/commands/chassis.start` and `snapcraft/commands/switch.start`.

Would it be more consistent to drop all of these simultaneously in a separate parent commit?


For the record, there are now some additional `DENIED` messages logged, but they do not affect operation.
```
[  101.595042] audit: type=1400 audit(1697770466.057:241): apparmor="DENIED" operation="exec" profile="snap.microovn.ovn-ovsdb-server-nb" name="/usr/bin/systemctl" pid=2804 comm="ovn-ctl" requested_mask="x" denied_mask="x" fsuid=0 ouid=0
[  101.595086] audit: type=1400 audit(1697770466.057:242): apparmor="DENIED" operation="exec" profile="snap.microovn.ovn-ovsdb-server-nb" name="/usr/bin/systemctl" pid=2804 comm="ovn-ctl" requested_mask="x" denied_mask="x" fsuid=0 ouid=0
[  101.596966] audit: type=1400 audit(1697770466.061:243): apparmor="DENIED" operation="exec" profile="snap.microovn.ovn-ovsdb-server-nb" name="/usr/bin/plymouth" pid=2806 comm="ovn-ctl" requested_mask="x" denied_mask="x" fsuid=0 ouid=0
[  101.597023] audit: type=1400 audit(1697770466.061:244): apparmor="DENIED" operation="exec" profile="snap.microovn.ovn-ovsdb-server-nb" name="/usr/bin/plymouth" pid=2806 comm="ovn-ctl" requested_mask="x" denied_mask="x" fsuid=0 ouid=0
```

Even without this change there are still other unhandled `DENIED` rules logged, such as:
```
[  101.599507] audit: type=1400 audit(1697770466.065:245): apparmor="DENIED" operation="capable" profile="snap.microovn.switch" pid=2787 comm="ovsdb-server" capability=38  capname="perfmon"
```

So I still think we should still drop these hacks and figure out what indirect dependencies of the OVS/OVN ctl scripts end up calling the offending binaries and provide patches upstream at some point in the future.

Reply:
I agree. I'll drop those instances too. The commands do not work anyway, it's better to have the failures logged rather than silently failing.